iot/api/CloudIotSample/Program.cs

```diff
@@ -1376,6 +1491,209 @@ public static object ClearRegistry(string projectId, string cloudRegion, string
+        {
+            var detachTopic = $"/devices/{deviceId}/detach";
+            Console.WriteLine("Detaching: {0}", detachTopic);
+            var BinaryData = Encoding.UTF8.GetBytes("{}");
+            client.Publish(detachTopic, BinaryData, MqttMsgBase.QOS_LEVEL_AT_LEAST_ONCE, true);
+            System.Threading.Thread.Sleep(2000);
+            return 0;
+        }
+        //[END iot_detach_device]
+
+        //[START iot_listen_for_config_messages]
+        public static object ListenForConfigMessages(string projectId, string cloudRegion,
+            string registryId, string deviceId, string gatewayId, int numMessages,
+            string privateKeyFile, string algorithm, string caCerts, string mqttBridgeHostname,
+            int mqttBridgePort, int jwtExpiresMinutes, int duration)
+        {
+            var clientId = $"projects/{projectId}/locations/{cloudRegion}/registries/{registryId}" +
+                $"/devices/{gatewayId}";
+            // Listens for configuration and system error messages on the gateway and
+            // bound devices.
+
+            var jwtIatTime = SystemClock.Instance.GetCurrentInstant();
+            // Create a duration
+            Duration durationExp = Duration.FromMinutes(jwtExpiresMinutes);
+            var jwtExpTime = SystemClock.Instance.GetCurrentInstant().Plus(durationExp);
+            var pass = CloudIotMqttExample.CreateJwtRsa(projectId, privateKeyFile);
+
+            // Use gateway to connect server
+            var mqttClient = StartMqtt(mqttBridgeHostname,
+                mqttBridgePort,
+                projectId,
+                cloudRegion,
+                registryId,
+                gatewayId,
+                privateKeyFile,
+                caCerts,
+                algorithm,
+                pass);
+
+            AttachDevice(mqttClient, deviceId);
+            // The topic devices receive configuration updates on.
+            string deviceConfigTopic = $"/devices/{deviceId}/config";
+
+            mqttClient.Subscribe(
+                new string[] { deviceConfigTopic },
+                new byte[] { MqttMsgBase.QOS_LEVEL_AT_LEAST_ONCE }
+            );
+
+            // Wait for about a minute for config messages.
+            for (int i = 0; i < duration; ++i)
+            {
+                Console.WriteLine("Listening...");
+                var secSinceIssue = SystemClock.Instance.GetCurrentInstant().Minus(jwtIatTime);
+                if (secSinceIssue.TotalSeconds > (60 * jwtExpiresMinutes)
+                    || !mqttClient.IsConnected)
+                {
+                    Console.WriteLine("Refreshing token after {0}s", secSinceIssue);
+                    jwtIatTime = SystemClock.Instance.GetCurrentInstant();
+                    mqttClient = CloudIotMqttExample.GetClient(projectId, cloudRegion,
+                        registryId, deviceId, privateKeyFile, algorithm,
+           caCerts, mqttBridgeHostname, mqttBridgePort);
```

Comment:
This continuation looks a little funny to me, maybe insert some whitespace before `caCerts` to match the previous line